doc/neps/nep-0034.rst

```diff
@@ -0,0 +1,99 @@
+===========================================================
+NEP 34 — Disallow inferring ``dtype=object`` from sequences
+===========================================================
+
+:Author: Matti Picus
+:Status: Draft
+:Type: Standards Track
+:Created: 2019-10-10
+
+
+Abstract
+--------
+
+``np.array([<sequence>, <sequence>])`` with no ``dtype`` keyword argument will
+sometimes default to an ``object``-dtype array. Change the behaviour to raise a
```

Comment:
Perhaps better as
```
``np.array(<ragged_nested_sequence>)`` with no ``dtype`` keyword argument will ...
```
or

```
``np.array(<nested_sequence>)``, when ``<nested_sequence>`` contains "ragged" sequences of different lengths, will when called with no ``dtype`` keyword argument ...
```

Reply:
Perhaps just clarify a bit more in plain language, e.g. add "if sequences are not all the same length". The word "ragged" is needlessly confusing.